sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubAuthenticationPolicy_token.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Azure.Core.Pipeline;
+using Microsoft.IdentityModel.Tokens;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary>
+    /// Web PubSub Authentication Policy.
+    /// </summary>
+    internal partial class WebPubSubAuthenticationPolicy : HttpPipelineSynchronousPolicy
+    {
+        private const int MaxTokenLength = 4096;
+
+        /// <summary>
+        /// Generates client bearwer token.
+        /// </summary>
+        /// <param name="audience"></param>
+        /// <param name="claims"></param>
+        /// <param name="key"></param>
+        /// <param name="expireAfter"></param>
+        /// <param name="hmacSha512">SHA512 if true, otherwise SHA256</param>
+        /// <returns></returns>
+        public static string GenerateAccessToken(
+            string audience,
+            IEnumerable<Claim> claims,
+            AzureKeyCredential key,
+            TimeSpan expireAfter = default,
+            bool hmacSha512 = false)
+        {
+            if (expireAfter == default) expireAfter = TimeSpan.FromMinutes(10);
```

Comment:
Here's another magical 10 minutes.  We should either compute these using RetryOptions or put something on our ClientOptions so it's configurable.